applications/tari_console_wallet/src/ui/state/app_state.rs

```diff
@@ -852,10 +861,72 @@ impl AppStateInner {
     }
 }
 
+#[derive(Clone)]
+pub struct CompletedTransactionInfo {
+    pub tx_id: TxId,
+    pub source_public_key: CommsPublicKey,
```

Comment:
Can reduce memory usage even more by storing the text representation instead of the crypto primitive
```suggestion
    pub source_public_key: String,
```